x-pack/packages/security-solution/common/src/cells/renderers/host/with_expandable_flyout.tsx

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, useMemo } from 'react';
+import { getFieldValue } from '@kbn/discover-utils';
+import type { PropsWithChildren } from 'react';
+import type { DataGridCellValueElementProps } from '@kbn/unified-data-table';
+import {
+  ExpandableFlyout,
+  type ExpandableFlyoutProps,
+  useExpandableFlyoutApi,
+  withExpandableFlyoutProvider,
+} from '@kbn/expandable-flyout';
+import { HostRightPanel, HostRightPanelProps } from '../../../flyout/panels';
+import { HostDetailsButton } from './button';
+
+export type HostCellWithFlyoutRendererProps = PropsWithChildren<DataGridCellValueElementProps>;
+
+const HostCellWithFlyoutRendererComp = React.memo(function HostCellWithFlyoutRendererComp(
+  props: HostCellWithFlyoutRendererProps
+) {
+  const hostName = getFieldValue(props.row, 'host.name');
+
+  const { openFlyout } = useExpandableFlyoutApi();
+
+  const onClick = useCallback(() => {
+    openFlyout({
+      right: {
+        id: `host-panel-${hostName}-${props.rowIndex}`,
+        params: {
+          hostName,
+        },
+      } as HostRightPanelProps,
+    });
+  }, [openFlyout, hostName, props.rowIndex]);
+
+  const panels: ExpandableFlyoutProps['registeredPanels'] = useMemo(() => {
+    return [
+      {
+        key: `host-panel-${hostName}-${props.rowIndex}`,
```

Comment:
I'm curious why duplicating each instance separately.. is it possible to wrap the table inside the flyout provider (ln64) instead? then it should only have 1 flyout

Reply:
Actually It is not possible because the `table` is owned by discover and we cannot add Context Providers that is why I went this way. 

I am also trying to comeup with the disadvantages of this approach but so far it feels okay. But i am open to more suggestions..